3520/1/lib/topgems.rb

```diff
@@ -0,0 +1,80 @@
+require './lib/repo_scrapper'
+require './lib/rubygems_links'
+require './lib/tg_parse'
+require './lib/output_table'
+require './lib/backup'
+require './lib/option_validation'
+
+class TopGems
+  include Backup
+  attr_reader :data
+
+  def initialize
+    @rg_links = RubyGemsLink.new
+    @data = []
+    @options = TGParse.parse
+    @scraper = RepoScrapper.new
+  end
+
+  def parse_all_links
+    @rg_links.yaml_links.each do |link|
+      @scraper.get_repo_page(link)
+      repo_info = @scraper.repo_info
+      Backup.backup_create(repo_info[:name], repo_info)
+    end
+  end
+
+  def gathering_data
+    @rg_links.gems_name.each do |name|
+      @data << Backup.backup_load(name)
+    end
+  end
+
+  def data_sort
+    @data.sort_by! { |el| el[:stars] }.reverse
+  end
+
+  def base_table
```

Comment:
what is the sense?

Reply:
My apology for that. I'm already move data_sort functional to base_table.